src/components/app/index.tsx

```diff
@@ -23,13 +25,31 @@ export default class App extends Component<Props, State> {
     }
   }
 
+  private async getImageData(bitmap: ImageBitmap): Promise<ImageData> {
```

Comment:
This doesn't read/write any class state, so it'd be better as a function.

Reply:
Right, good call. I should probably start a `imageutils` file or something. We are gonna write a lot more of these I reckon.

Reply:
Also, we should do this on the worker side if we can (will be fun to play with offscreen canvas)

Reply:
`OffscreenCanvas` isn't stable anywhere tho.

Reply:
... could we built an "Offthread" canvas  with an iframe? Cc @developit

Reply:
I guess we should see how much jank it causes us first.

The iframe idea is interesting! OOPIFs are a thing now, so it could work. I wonder if transferrables lose some of their benefit when jumping across processes.